# Random Forests classification attempt for EFD and EWT

## 1. Generate 48*1 vectors

In [1]:
import pandas as pd
import numpy as np

In [2]:
df=pd.read_csv(r'C:\Ninapro\New folder\10SUB17GES12CHNL_V2.csv')

In [3]:
df.head()

Source      EWT_1      EWT_2      EWT_3     EWT_4     EWT_5  \
0  _S10_C10_G10_T1  42.393391  20.353574  22.221536  4.677523  8.101762   
1  _S10_C10_G10_T2  40.307319  22.946148  22.192473  4.043746  8.488849   
2  _S10_C10_G10_T3  63.723407  17.697616   6.197031  3.495138  7.814109   
3  _S10_C10_G10_T4  73.270844   3.967913   8.299588  4.693181  8.317369   
4  _S10_C10_G10_T5  50.725586  19.402604  19.435229  3.148787  6.172383   

   EWT Comp 90%  EWT Comp 95%     EFD_1      EFD_2      EFD_3      EFD_4  \
0             5             5  1.869982   7.248724   4.363911  37.067306   
1             5             5  7.266559  22.428292   8.308186  14.190720   
2             5             5  6.535971   7.370917  30.548617   6.384719   
3             5             5  0.000000   2.205047   5.608725  10.609532   
4             5             5  4.052417  38.718434   6.081567   2.494551   

       EFD_5  EFD Comp 90%  EFD Comp 95%  
0  50.083143             4             5  
1  47.024652             5             5  
2  48.297447             5             5  
3  82.406822             3             4  
4  48.324557             4             5

In [4]:
L=df['Source'].str[1:]
df['source']=pd.DataFrame(L)
a=df[['EWT_1','EWT_2','EWT_3','EWT_4','EWT_5']]
b=df[['EFD_1','EFD_2','EFD_3','EFD_4','EFD_5']]

In [5]:
li0=[]
liindex=[]
for i in range(len(a)):
    e=a.iloc[i,:].sort_values(ascending=False)[:4]
    
    remaining=[]
    for j in range(4):
        remaining.append(e.index[j][-1])
    liindex.append(remaining)
    
    thr=0
    nochange=[]
    for k in range(4):
        if thr<90:
            thr=thr+e[k]
            nochange.append(e.index[k][-1])
        else:
            nochange.append(e.index[k][-1])
            break
    li0.append(nochange)

In [6]:
li1=[]
liindex1=[]
for i in range(len(b)):
    e=b.iloc[i,:].sort_values(ascending=False)[:4]
    
    remaining=[]
    for j in range(4):
        remaining.append(e.index[j][-1])
    liindex1.append(remaining)
    
    thr=0
    nochange=[]
    for k in range(4):
        if thr<90:
            thr=thr+e[k]
            nochange.append(e.index[k][-1])
        else:
            nochange.append(e.index[k][-1])
            break
    li1.append(nochange)

In [7]:
df_d=pd.read_csv(r'C:\Ninapro\dfewt.csv')
df_f=pd.read_csv(r'C:\Ninapro\dfefd_full.csv')

In [8]:
df_ful_efd=pd.merge(df,df_f, how='left')

In [9]:
df_ful_efd

Source      EWT_1      EWT_2      EWT_3      EWT_4      EWT_5  \
0      _S10_C10_G10_T1  42.393391  20.353574  22.221536   4.677523   8.101762   
1      _S10_C10_G10_T2  40.307319  22.946148  22.192473   4.043746   8.488849   
2      _S10_C10_G10_T3  63.723407  17.697616   6.197031   3.495138   7.814109   
3      _S10_C10_G10_T4  73.270844   3.967913   8.299588   4.693181   8.317369   
4      _S10_C10_G10_T5  50.725586  19.402604  19.435229   3.148787   6.172383   
...                ...        ...        ...        ...        ...        ...   
12235     _S9_C9_G9_T2  29.619568  14.894297   5.577779   4.845001  42.284773   
12236     _S9_C9_G9_T3  47.318324   9.887580  10.818323   9.612869  16.981519   
12237     _S9_C9_G9_T4  65.372635  10.292249   4.537400   7.673387   9.707764   
12238     _S9_C9_G9_T5  65.392173   4.101990   4.150240   5.904328  18.488868   
12239     _S9_C9_G9_T6  42.828271   3.550446  11.629725  14.434107  26.009291   

       EWT Comp 90%  EWT Comp 95%      EFD_1      EFD_2  ...      EFD_4  \
0                 5             5   1.869982   7.248724  ...  37.067306   
1                 5             5   7.266559  22.428292  ...  14.190720   
2                 5             5   6.535971   7.370917  ...   6.384719   
3                 5             5   0.000000   2.205047  ...  10.609532   
4                 5             5   4.052417  38.718434  ...   2.494551   
...             ...           ...        ...        ...  ...        ...   
12235             5             5  23.154214  25.061234  ...  15.527512   
12236             5             5  11.478585   3.166186  ...   5.174070   
12237             5             5   9.865960   5.723378  ...   6.912941   
12238             5             5  20.323694   6.025932  ...   4.841017   
12239             5             5  22.840014   0.629758  ...  13.786692   

           EFD_5  EFD Comp 90%  EFD Comp 95%          source         1  \
0      50.083143             4             5  S10_C10_G10_T1  0.323496   
1      47.024652             5             5  S10_C10_G10_T2  0.364284   
2      48.297447             5             5  S10_C10_G10_T3  0.326980   
3      82.406822             3             4  S10_C10_G10_T4  0.000000   
4      48.324557             4             5  S10_C10_G10_T5  0.336663   
...          ...           ...           ...             ...       ...   
12235  24.709736             5             5     S9_C9_G9_T2  0.241919   
12236  72.390154             4             5     S9_C9_G9_T3  0.258353   
12237  71.338811             5             5     S9_C9_G9_T4  0.254900   
12238  65.117253             4             5     S9_C9_G9_T5  0.245945   
12239  56.944907             4             5     S9_C9_G9_T6  0.249851   

              2         3         4         5  
0      0.324724  0.284899  0.276373  0.150836  
1      0.293343  0.314072  0.277204  0.158509  
2      0.296051  0.283640  0.250479  0.139150  
3      0.361567  0.389624  0.408839  0.227059  
4      0.311441  0.251871  0.204114  0.153872  
...         ...       ...       ...       ...  
12235  0.349181  0.415429  0.387244  0.107003  
12236  0.235725  0.220736  0.232806  0.069459  
12237  0.247380  0.212512  0.206282  0.084053  
12238  0.209902  0.203999  0.209125  0.053028  
12239  0.147392  0.301953  0.390480  0.343677  

[12240 rows x 21 columns]

In [10]:
df_ful_ewt=pd.merge(df,df_d, how='left')

In [11]:
df_ful_ewt

Source      EWT_1      EWT_2      EWT_3      EWT_4      EWT_5  \
0      _S10_C10_G10_T1  42.393391  20.353574  22.221536   4.677523   8.101762   
1      _S10_C10_G10_T2  40.307319  22.946148  22.192473   4.043746   8.488849   
2      _S10_C10_G10_T3  63.723407  17.697616   6.197031   3.495138   7.814109   
3      _S10_C10_G10_T4  73.270844   3.967913   8.299588   4.693181   8.317369   
4      _S10_C10_G10_T5  50.725586  19.402604  19.435229   3.148787   6.172383   
...                ...        ...        ...        ...        ...        ...   
12235     _S9_C9_G9_T2  29.619568  14.894297   5.577779   4.845001  42.284773   
12236     _S9_C9_G9_T3  47.318324   9.887580  10.818323   9.612869  16.981519   
12237     _S9_C9_G9_T4  65.372635  10.292249   4.537400   7.673387   9.707764   
12238     _S9_C9_G9_T5  65.392173   4.101990   4.150240   5.904328  18.488868   
12239     _S9_C9_G9_T6  42.828271   3.550446  11.629725  14.434107  26.009291   

       EWT Comp 90%  EWT Comp 95%      EFD_1      EFD_2  ...      EFD_4  \
0                 5             5   1.869982   7.248724  ...  37.067306   
1                 5             5   7.266559  22.428292  ...  14.190720   
2                 5             5   6.535971   7.370917  ...   6.384719   
3                 5             5   0.000000   2.205047  ...  10.609532   
4                 5             5   4.052417  38.718434  ...   2.494551   
...             ...           ...        ...        ...  ...        ...   
12235             5             5  23.154214  25.061234  ...  15.527512   
12236             5             5  11.478585   3.166186  ...   5.174070   
12237             5             5   9.865960   5.723378  ...   6.912941   
12238             5             5  20.323694   6.025932  ...   4.841017   
12239             5             5  22.840014   0.629758  ...  13.786692   

           EFD_5  EFD Comp 90%  EFD Comp 95%          source         1  \
0      50.083143             4             5  S10_C10_G10_T1  0.139901   
1      47.024652             5             5  S10_C10_G10_T2  0.151750   
2      48.297447             5             5  S10_C10_G10_T3  0.153472   
3      82.406822             3             4  S10_C10_G10_T4  0.214029   
4      48.324557             4             5  S10_C10_G10_T5  0.156771   
...          ...           ...           ...             ...       ...   
12235  24.709736             5             5     S9_C9_G9_T2  0.125742   
12236  72.390154             4             5     S9_C9_G9_T3  0.054639   
12237  71.338811             5             5     S9_C9_G9_T4  0.082748   
12238  65.117253             4             5     S9_C9_G9_T5  0.053115   
12239  56.944907             4             5     S9_C9_G9_T6  0.299715   

              2         3         4         5  
0      0.258925  0.277124  0.321873  0.307725  
1      0.271766  0.295154  0.299055  0.350643  
2      0.287540  0.280703  0.322790  0.301014  
3      0.465753  0.538328  0.461133  0.305963  
4      0.268095  0.326295  0.347187  0.320904  
...         ...       ...       ...       ...  
12235  0.437652  0.429218  0.395859  0.245290  
12236  0.151945  0.168905  0.211165  0.250159  
12237  0.203101  0.206113  0.229784  0.255787  
12238  0.216842  0.200066  0.216329  0.249938  
12239  0.513797  0.426539  0.414341  0.247318  

[12240 rows x 21 columns]

In [12]:
org=['1', '3', '2', '5', '4']

for i in range(12240):
    m=tuple(list(set(org).difference(set(li0[i]))))
    df_ful_ewt.loc[i, m] = 0

In [13]:
org=['1', '3', '2', '5', '4']

for i in range(12240):
    m=tuple(list(set(org).difference(set(li1[i]))))
    df_ful_efd.loc[i, m] = 0

In [14]:
df_new_ewt=df_ful_ewt[['source','1','2','3','4','5']]
df_new_efd=df_ful_efd[['source','1','2','3','4','5']]

In [15]:
df_new_ewt

source         1         2         3         4         5
0      S10_C10_G10_T1  0.139901  0.258925  0.277124  0.000000  0.307725
1      S10_C10_G10_T2  0.151750  0.271766  0.295154  0.000000  0.350643
2      S10_C10_G10_T3  0.153472  0.287540  0.280703  0.000000  0.301014
3      S10_C10_G10_T4  0.214029  0.000000  0.538328  0.461133  0.305963
4      S10_C10_G10_T5  0.156771  0.268095  0.326295  0.000000  0.320904
...               ...       ...       ...       ...       ...       ...
12235     S9_C9_G9_T2  0.125742  0.437652  0.429218  0.000000  0.245290
12236     S9_C9_G9_T3  0.054639  0.151945  0.168905  0.000000  0.250159
12237     S9_C9_G9_T4  0.082748  0.203101  0.000000  0.229784  0.255787
12238     S9_C9_G9_T5  0.053115  0.000000  0.200066  0.216329  0.249938
12239     S9_C9_G9_T6  0.299715  0.000000  0.426539  0.414341  0.247318

[12240 rows x 6 columns]

In [16]:
df_new_efd

source         1         2         3         4         5
0      S10_C10_G10_T1  0.000000  0.324724  0.284899  0.276373  0.150836
1      S10_C10_G10_T2  0.000000  0.293343  0.314072  0.277204  0.158509
2      S10_C10_G10_T3  0.326980  0.296051  0.283640  0.000000  0.139150
3      S10_C10_G10_T4  0.000000  0.000000  0.389624  0.408839  0.227059
4      S10_C10_G10_T5  0.336663  0.311441  0.251871  0.000000  0.153872
...               ...       ...       ...       ...       ...       ...
12235     S9_C9_G9_T2  0.241919  0.349181  0.000000  0.387244  0.107003
12236     S9_C9_G9_T3  0.258353  0.000000  0.220736  0.232806  0.069459
12237     S9_C9_G9_T4  0.254900  0.000000  0.212512  0.206282  0.084053
12238     S9_C9_G9_T5  0.245945  0.209902  0.000000  0.209125  0.053028
12239     S9_C9_G9_T6  0.249851  0.000000  0.301953  0.390480  0.343677

[12240 rows x 6 columns]

In [17]:
import warnings
warnings.filterwarnings('ignore')

In [18]:
ewt48=pd.DataFrame()

for i in range(12240):
    diff=list(set(['1','2','3','4','5'])-set(liindex[i]))
    row=df_new_ewt.iloc[[i]].drop(diff,axis=1).iloc[0].values
    ewt48=ewt48.append(pd.DataFrame(row.reshape(1,-1)))

In [19]:
ewt48

0         1         2         3         4
0   S10_C10_G10_T1  0.139901  0.258925  0.277124  0.307725
0   S10_C10_G10_T2   0.15175  0.271766  0.295154  0.350643
0   S10_C10_G10_T3  0.153472   0.28754  0.280703  0.301014
0   S10_C10_G10_T4  0.214029  0.538328  0.461133  0.305963
0   S10_C10_G10_T5  0.156771  0.268095  0.326295  0.320904
..             ...       ...       ...       ...       ...
0      S9_C9_G9_T2  0.125742  0.437652  0.429218   0.24529
0      S9_C9_G9_T3  0.054639  0.151945  0.168905  0.250159
0      S9_C9_G9_T4  0.082748  0.203101  0.229784  0.255787
0      S9_C9_G9_T5  0.053115  0.200066  0.216329  0.249938
0      S9_C9_G9_T6  0.299715  0.426539  0.414341  0.247318

[12240 rows x 5 columns]

In [20]:
efd48=pd.DataFrame()

for i in range(12240):
    diff=list(set(['1','2','3','4','5'])-set(liindex1[i]))
    row=df_new_efd.iloc[[i]].drop(diff,axis=1).iloc[0].values
    efd48=efd48.append(pd.DataFrame(row.reshape(1,-1)))

In [21]:
efd48

0         1         2         3         4
0   S10_C10_G10_T1  0.324724  0.284899  0.276373  0.150836
0   S10_C10_G10_T2  0.293343  0.314072  0.277204  0.158509
0   S10_C10_G10_T3   0.32698  0.296051   0.28364   0.13915
0   S10_C10_G10_T4       0.0  0.389624  0.408839  0.227059
0   S10_C10_G10_T5  0.336663  0.311441  0.251871  0.153872
..             ...       ...       ...       ...       ...
0      S9_C9_G9_T2  0.241919  0.349181  0.387244  0.107003
0      S9_C9_G9_T3  0.258353  0.220736  0.232806  0.069459
0      S9_C9_G9_T4    0.2549  0.212512  0.206282  0.084053
0      S9_C9_G9_T5  0.245945  0.209902  0.209125  0.053028
0      S9_C9_G9_T6  0.249851  0.301953   0.39048  0.343677

[12240 rows x 5 columns]

In [22]:
efd48.to_csv(r'C:\Ninapro\efd48.csv',index=False)
ewt48.to_csv(r'C:\Ninapro\ewt48.csv',index=False)

In [5]:
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score
import matplotlib.pyplot as plt

## 2. Three gestures classification
### 60*1 input vectors EWT

In [2]:
tr1=pd.read_csv(r'C:\Ninapro\ewtvect60.csv')

In [25]:
#tr1['cls']=tr1['Gesture'].str[1:]

In [3]:
tr1

g  Unnamed: 1  Unnamed: 2  Unnamed: 3  Unnamed: 4  Unnamed: 5  \
0    1    0.250072    0.159295    0.185298    0.222070    0.266288   
1    1    0.267416    0.123359    0.144753    0.178831    0.253865   
2    1    0.250811    0.133088    0.000000    0.191810    0.280505   
3    1    0.243922    0.147095    0.184018    0.229810    0.288911   
4    1    0.233361    0.166396    0.243245    0.309512    0.297370   
..  ..         ...         ...         ...         ...         ...   
475  9    0.365947    0.249758    0.272549    0.274420    0.256113   
476  9    0.349102    0.241113    0.246334    0.259333    0.254964   
477  9    0.420407    0.337403    0.305792    0.240617    0.231203   
478  9    0.383993    0.289823    0.236253    0.281943    0.226182   
479  9    0.459857    0.431279    0.354357    0.287700    0.247016   

     Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  ...  Unnamed: 51  \
0      0.222324    0.289427    0.300398    0.324126  ...     0.060121   
1      0.192772    0.240966    0.268960    0.284218  ...     0.065451   
2      0.202640    0.264511    0.263447    0.311064  ...     0.062505   
3      0.223307    0.299277    0.300186    0.324850  ...     0.056973   
4      0.211741    0.258573    0.284544    0.287395  ...     0.064544   
..          ...         ...         ...         ...  ...          ...   
475    0.362897    0.430416    0.415659    0.346144  ...     0.626105   
476    0.478350    0.559422    0.574407    0.598261  ...     0.624424   
477    0.375903    0.441267    0.390082    0.436166  ...     0.578162   
478    0.329254    0.444118    0.436802    0.400667  ...     0.512791   
479    0.505600    0.000000    0.559359    0.518821  ...     0.618723   

     Unnamed: 52  Unnamed: 53  Unnamed: 54  Unnamed: 55  Unnamed: 56  \
0       0.245680     0.256508     0.000000     0.275563     0.164559   
1       0.000000     0.257068     0.265752     0.278540     0.151206   
2       0.243944     0.255973     0.000000     0.270985     0.203783   
3       0.289988     0.000000     0.278398     0.273834     0.312832   
4       0.000000     0.000000     0.270558     0.273835     0.189563   
..           ...          ...          ...          ...          ...   
475     0.670014     0.612990     0.641517     0.519933     0.075959   
476     0.613665     0.591229     0.683225     0.577567     0.081818   
477     0.614691     0.661999     0.629274     0.542997     0.073700   
478     0.577019     0.642699     0.608191     0.531237     0.090233   
479     0.600276     0.637344     0.000000     0.596692     0.138756   

     Unnamed: 57  Unnamed: 58  Unnamed: 59  Unnamed: 60  
0       0.173895     0.194899     0.208562     0.236200  
1       0.159693     0.169872     0.178787     0.221973  
2       0.173311     0.186488     0.207479     0.249848  
3       0.264680     0.249200     0.268134     0.278013  
4       0.174602     0.198010     0.215739     0.248095  
..           ...          ...          ...          ...  
475     0.203738     0.223829     0.236920     0.265187  
476     0.217231     0.222425     0.251125     0.259508  
477     0.206247     0.250791     0.263824     0.255186  
478     0.239933     0.270279     0.000000     0.250284  
479     0.341398     0.265603     0.000000     0.256309  

[480 rows x 61 columns]

In [6]:
x = tr1.drop('g',axis=1)
y= tr1['g']
#x = tr1.drop(['Index','Subject','Gesture','Rep','cls'],axis=1)
#y = tr1['cls']
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.25)

In [7]:
from sklearn.model_selection import KFold, cross_val_score
from sklearn.ensemble import RandomForestClassifier

In [8]:
rfc = RandomForestClassifier()
scores_rfc_acc = cross_val_score(rfc, x_train, y_train, cv=5, scoring='accuracy')
scores_rfc_recall = cross_val_score(rfc, x_train, y_train, cv=5, scoring='recall_macro')
scores_rfc_pre = cross_val_score(rfc, x_train, y_train, cv=5, scoring='precision_macro')

In [9]:
print (scores_rfc_acc.mean())
print (scores_rfc_recall.mean())
print (scores_rfc_pre.mean())

0.7222222222222222
0.7247411616161618
0.7212905497280497


### 60*1 input vectors EFD

In [10]:
tr2=pd.read_csv(r'C:\Ninapro\efdvect60.csv')

In [11]:
tr2

g  Unnamed: 1  Unnamed: 2  Unnamed: 3  Unnamed: 4  Unnamed: 5  \
0    1    0.243920    0.139569    0.000000    0.123496    0.197000   
1    1    0.240870    0.150842    0.117058    0.088571    0.220013   
2    1    0.277644    0.182200    0.164350    0.119442    0.211351   
3    1    0.290810    0.227068    0.175867    0.148176    0.230032   
4    1    0.297343    0.000000    0.296390    0.192565    0.251840   
..  ..         ...         ...         ...         ...         ...   
475  9    0.254032    0.265771    0.257317    0.000000    0.360080   
476  9    0.248635    0.000000    0.000000    0.258858    0.274440   
477  9    0.237695    0.000000    0.238014    0.244295    0.300115   
478  9    0.238156    0.261400    0.240945    0.275535    0.338941   
479  9    0.248471    0.000000    0.260893    0.328203    0.429245   

     Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  ...  Unnamed: 51  \
0      0.335933    0.316487    0.292910    0.281642  ...     0.273460   
1      0.336167    0.315668    0.000000    0.266755  ...     0.279222   
2      0.307864    0.331213    0.288124    0.265302  ...     0.267515   
3      0.336283    0.341084    0.317018    0.000000  ...     0.282087   
4      0.337752    0.327508    0.286805    0.272709  ...     0.277722   
..          ...         ...         ...         ...  ...          ...   
475    0.286348    0.000000    0.246069    0.434409  ...     0.535076   
476    0.326530    0.546891    0.468735    0.000000  ...     0.559721   
477    0.272771    0.429542    0.388291    0.406997  ...     0.543136   
478    0.288636    0.000000    0.388077    0.438737  ...     0.546218   
479    0.291621    0.511052    0.492242    0.489982  ...     0.602795   

     Unnamed: 52  Unnamed: 53  Unnamed: 54  Unnamed: 55  Unnamed: 56  \
0       0.263330     0.000000     0.234093     0.053007     0.244873   
1       0.268330     0.000000     0.254658     0.066274     0.220903   
2       0.256470     0.000000     0.000000     0.063148     0.251346   
3       0.261294     0.321002     0.258620     0.057548     0.281392   
4       0.268779     0.000000     0.000000     0.066561     0.237762   
..           ...          ...          ...          ...          ...   
475     0.654981     0.624153     0.527202     0.568301     0.246794   
476     0.583211     0.620033     0.633872     0.627649     0.254999   
477     0.474250     0.501953     0.622699     0.575598     0.255049   
478     0.619839     0.318894     0.494163     0.504678     0.246499   
479     0.627253     0.585358     0.591856     0.611973     0.254269   

     Unnamed: 57  Unnamed: 58  Unnamed: 59  Unnamed: 60  
0       0.226393     0.214620     0.198087     0.177696  
1       0.177334     0.162058     0.148536     0.145744  
2       0.218027     0.192291     0.000000     0.214860  
3       0.267393     0.244147     0.251347     0.283137  
4       0.189328     0.185484     0.000000     0.186116  
..           ...          ...          ...          ...  
475     0.233767     0.225055     0.216062     0.081590  
476     0.218826     0.199676     0.196305     0.069999  
477     0.000000     0.260925     0.221515     0.089954  
478     0.000000     0.226973     0.198753     0.087027  
479     0.000000     0.349335     0.307049     0.141197  

[480 rows x 61 columns]

In [12]:
x1 = tr2.drop('g',axis=1)
y1 = tr2['g']
x1_train, x1_test, y1_train, y1_test = train_test_split(x1, y1, test_size=0.25)

In [13]:
rfc_1 = RandomForestClassifier()
scores_rfc_acc1 = cross_val_score(rfc_1, x1_train, y1_train, cv=5, scoring='accuracy')
scores_rfc_recall1 = cross_val_score(rfc_1, x1_train, y1_train, cv=5, scoring='recall_macro')
scores_rfc_pre1 = cross_val_score(rfc_1, x1_train, y1_train, cv=5, scoring='precision_macro')

In [14]:
print (scores_rfc_acc1.mean())
print (scores_rfc_recall1.mean())
print (scores_rfc_pre1.mean())

0.6944444444444444
0.6880555555555555
0.6814581252081252


### 48*1 input vectors EWT

In [15]:
tr3=pd.read_csv(r'C:\Ninapro\ewtvect48.csv')
tr3

g  Unnamed: 1  Unnamed: 2  Unnamed: 3  Unnamed: 4  Unnamed: 5  \
0    1    0.139901    0.258925    0.277124    0.307725    0.087963   
1    1    0.151750    0.271766    0.295154    0.350643    0.087206   
2    1    0.153472    0.287540    0.280703    0.301014    0.091119   
3    1    0.214029    0.538328    0.461133    0.305963    0.075418   
4    1    0.156771    0.268095    0.326295    0.320904    0.076773   
..  ..         ...         ...         ...         ...         ...   
475  9    0.088794    0.203192    0.255605    0.243050    0.228492   
476  9    0.069176    0.214425    0.215461    0.255682    0.208125   
477  9    0.039473    0.145638    0.179855    0.227246    0.212531   
478  9    0.064485    0.204721    0.241145    0.267269    0.173120   
479  9    0.068391    0.170524    0.202463    0.230835    0.259207   

     Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  ...  Unnamed: 39  \
0      0.141892    0.185236    0.244481    0.086805  ...     0.323982   
1      0.172734    0.214287    0.248407    0.088794  ...     0.302720   
2      0.176715    0.198736    0.232748    0.073691  ...     0.382305   
3      0.131554    0.169363    0.246548    0.075663  ...     0.408825   
4      0.140502    0.187909    0.242850    0.084377  ...     0.384379   
..          ...         ...         ...         ...  ...          ...   
475    0.260231    0.252509    0.251903    0.126833  ...     0.257973   
476    0.314592    0.286839    0.258910    0.411442  ...     0.259808   
477    0.275173    0.268056    0.262707    0.191682  ...     0.362193   
478    0.267993    0.238201    0.264460    0.099968  ...     0.270761   
479    0.266198    0.287166    0.260592    0.176337  ...     0.231441   

     Unnamed: 40  Unnamed: 41  Unnamed: 42  Unnamed: 43  Unnamed: 44  \
0       0.336824     0.082017     0.246220     0.245549     0.255302   
1       0.335337     0.083121     0.309628     0.259957     0.253651   
2       0.357327     0.089390     0.227949     0.262494     0.262902   
3       0.410428     0.111388     0.303379     0.278543     0.262799   
4       0.362741     0.093169     0.279220     0.276215     0.257474   
..           ...          ...          ...          ...          ...   
475     0.260028     0.185536     0.258571     0.275523     0.282629   
476     0.245716     0.196294     0.349890     0.000000     0.305847   
477     0.246677     0.179401     0.000000     0.258711     0.272464   
478     0.261875     0.204223     0.285175     0.280176     0.297341   
479     0.256394     0.228922     0.376433     0.345949     0.370363   

     Unnamed: 45  Unnamed: 46  Unnamed: 47  Unnamed: 48  
0       0.415130     0.326726     0.387825     0.429353  
1       0.405720     0.320378     0.382998     0.407870  
2       0.390904     0.336744     0.327910     0.382950  
3       0.420446     0.358909     0.411342     0.412316  
4       0.395827     0.298055     0.336802     0.373278  
..           ...          ...          ...          ...  
475     0.146323     0.095968     0.129226     0.183492  
476     0.151101     0.114663     0.156044     0.315201  
477     0.130909     0.068560     0.120631     0.194584  
478     0.167168     0.110246     0.157430     0.231905  
479     0.182418     0.148003     0.206347     0.423552  

[480 rows x 49 columns]

In [23]:
x3 = tr3.drop('g',axis=1)
y3 = tr3['g']
x3_train, x3_test, y3_train, y3_test = train_test_split(x3, y3, test_size=0.25)

In [24]:
rfc_3 = RandomForestClassifier()
scores_rfc_acc3 = cross_val_score(rfc_3, x3_train, y3_train, cv=5, scoring='accuracy')
scores_rfc_recall3 = cross_val_score(rfc_3, x3_train, y3_train, cv=5, scoring='recall_macro')
scores_rfc_pre3 = cross_val_score(rfc_3, x3_train, y3_train, cv=5, scoring='precision_macro')

In [25]:
print (scores_rfc_acc3.mean())
print (scores_rfc_recall3.mean())
print (scores_rfc_pre3.mean())

0.9833333333333334
0.9819444444444445
0.985909090909091


### 48*1 input vectors EFD

In [19]:
tr4=pd.read_csv(r'C:\Ninapro\efdvect48.csv')
tr4

g  Unnamed: 1  Unnamed: 2  Unnamed: 3  Unnamed: 4  Unnamed: 5  \
0    1    0.243920    0.139569    0.123496    0.197000    0.335933   
1    1    0.240870    0.150842    0.117058    0.220013    0.336167   
2    1    0.277644    0.164350    0.119442    0.211351    0.307864   
3    1    0.290810    0.227068    0.175867    0.230032    0.336283   
4    1    0.297343    0.296390    0.192565    0.251840    0.327508   
..  ..         ...         ...         ...         ...         ...   
475  9    0.254032    0.265771    0.257317    0.360080    0.286348   
476  9    0.248635    0.000000    0.258858    0.274440    0.326530   
477  9    0.237695    0.238014    0.244295    0.300115    0.272771   
478  9    0.238156    0.240945    0.275535    0.338941    0.288636   
479  9    0.248471    0.260893    0.328203    0.429245    0.291621   

     Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  ...  Unnamed: 39  \
0      0.316487    0.281642    0.217952    0.317944  ...     0.202050   
1      0.315668    0.266755    0.197995    0.247416  ...     0.289189   
2      0.288124    0.265302    0.200980    0.270949  ...     0.278398   
3      0.341084    0.317018    0.223835    0.253441  ...     0.230273   
4      0.286805    0.272709    0.209465    0.299897  ...     0.268422   
..          ...         ...         ...         ...  ...          ...   
475    0.246069    0.434409    0.360967    0.237873  ...     0.263971   
476    0.546891    0.468735    0.513400    0.294714  ...     0.256139   
477    0.429542    0.406997    0.363951    0.247544  ...     0.000000   
478    0.388077    0.438737    0.353168    0.250531  ...     0.245065   
479    0.511052    0.489982    0.501494    0.479476  ...     0.000000   

     Unnamed: 40  Unnamed: 41  Unnamed: 42  Unnamed: 43  Unnamed: 44  \
0       0.205384     0.273460     0.263330     0.234093     0.053007   
1       0.285442     0.279222     0.268330     0.254658     0.066274   
2       0.235677     0.267515     0.256470     0.000000     0.063148   
3       0.290879     0.282087     0.261294     0.258620     0.057548   
4       0.297964     0.277722     0.268779     0.000000     0.066561   
..           ...          ...          ...          ...          ...   
475     0.134439     0.535076     0.654981     0.527202     0.568301   
476     0.125050     0.559721     0.620033     0.633872     0.627649   
477     0.129933     0.543136     0.474250     0.622699     0.575598   
478     0.135017     0.546218     0.619839     0.494163     0.504678   
479     0.126229     0.602795     0.627253     0.585358     0.611973   

     Unnamed: 45  Unnamed: 46  Unnamed: 47  Unnamed: 48  
0       0.244873     0.214620     0.198087     0.177696  
1       0.220903     0.177334     0.162058     0.145744  
2       0.251346     0.218027     0.192291     0.214860  
3       0.281392     0.267393     0.251347     0.283137  
4       0.237762     0.189328     0.185484     0.186116  
..           ...          ...          ...          ...  
475     0.246794     0.225055     0.216062     0.081590  
476     0.254999     0.218826     0.199676     0.069999  
477     0.255049     0.260925     0.221515     0.089954  
478     0.246499     0.226973     0.198753     0.087027  
479     0.254269     0.349335     0.307049     0.141197  

[480 rows x 49 columns]

In [20]:
x4 = tr4.drop('g',axis=1)
y4 = tr4['g']
x4_train, x4_test, y4_train, y4_test = train_test_split(x4, y4, test_size=0.25)

In [21]:
rfc_4 = RandomForestClassifier()
scores_rfc_acc4 = cross_val_score(rfc_4, x4_train, y4_train, cv=5, scoring='accuracy')
scores_rfc_recall4 = cross_val_score(rfc_4, x4_train, y4_train, cv=5, scoring='recall_macro')
scores_rfc_pre4 = cross_val_score(rfc_4, x4_train, y4_train, cv=5, scoring='precision_macro')

In [22]:
print (scores_rfc_acc4.mean())
print (scores_rfc_recall4.mean())
print (scores_rfc_pre4.mean())

0.7694444444444445
0.7511111111111111
0.769325604950605


### Original Data without Decomposition

In [31]:
tr5=pd.read_csv(r'C:\Ninapro\orgvect12.csv')
tr5

g  Unnamed: 1  Unnamed: 2  Unnamed: 3  Unnamed: 4  Unnamed: 5  \
0     1    0.314624    0.243366    0.313086    0.104637    0.084175   
1     1    0.390897    0.238802    0.174411    0.080822    0.077375   
2     1    0.336415    0.228998    0.263962    0.081919    0.084285   
3     1    0.311128    0.245574    0.173392    0.084604    0.078758   
4     1    0.304438    0.238745    0.286031    0.086237    0.080482   
...  ..         ...         ...         ...         ...         ...   
1015  9    0.253504    0.329751    0.223974    0.134360    0.119203   
1016  9    0.236662    0.457381    0.301265    0.130417    0.097749   
1017  9    0.230982    0.350147    0.290974    0.132783    0.121680   
1018  9    0.232303    0.334373    0.325980    0.128972    0.102102   
1019  9    0.279870    0.463808    0.625462    0.124953    0.102812   

      Unnamed: 6  Unnamed: 7  Unnamed: 8  Unnamed: 9  Unnamed: 10  \
0       0.097428    0.213761    0.265235    0.146477     0.229233   
1       0.088403    0.305054    0.237242    0.072786     0.278142   
2       0.087823    0.263912    0.271545    0.072505     0.280162   
3       0.095461    0.280793    0.163820    0.104666     0.292360   
4       0.092356    0.272996    0.197009    0.097716     0.294568   
...          ...         ...         ...         ...          ...   
1015    0.124659    0.457794    0.413996    0.187973     0.142480   
1016    0.112684    0.361205    0.311121    0.078962     0.140461   
1017    0.112933    0.355486    0.271502    0.090007     0.139251   
1018    0.111453    0.361605    0.346729    0.076676     0.134171   
1019    0.102877    0.334291    0.371730    0.249542     0.132243   

      Unnamed: 11  Unnamed: 12  
0        0.074493     0.240686  
1        0.077994     0.245581  
2        0.079974     0.293041  
3        0.079216     0.371716  
4        0.075325     0.280009  
...           ...          ...  
1015     0.645580     0.108548  
1016     0.650997     0.106866  
1017     0.595291     0.114692  
1018     0.551090     0.123197  
1019     0.624256     0.161752  

[1020 rows x 13 columns]

In [32]:
x5 = tr5.drop('g',axis=1)
y5 = tr5['g']
x5_train, x5_test, y5_train, y5_test = train_test_split(x5, y5, test_size=0.25)

In [33]:
rfc_5 = RandomForestClassifier()
scores_rfc_acc5 = cross_val_score(rfc_5, x5_train, y5_train, cv=5, scoring='accuracy')
scores_rfc_recall5 = cross_val_score(rfc_5, x5_train, y5_train, cv=5, scoring='recall_macro')
scores_rfc_pre5 = cross_val_score(rfc_5, x5_train, y5_train, cv=5, scoring='precision_macro')

In [34]:
print (scores_rfc_acc5.mean())
print (scores_rfc_recall5.mean())
print (scores_rfc_pre5.mean())

0.7581699346405228
0.7515300059417707
0.7698082962788846


| Metric | EWT60 | EFD60 | EWT48 | EFD48 | ORG12 |
| --- | --- | --- | --- | --- | --- |
| Accuracy | 0.722 | 0.694|	0.983 |	0.769 | 0.758
| Recall | 0.725 |	0.688 |	0.982 |	0.751 | 0.752
| Precision | 0.721 | 0.681 |	0.986 |	0.769 | 0.770